configure.py

```diff
@@ -0,0 +1,137 @@
+  parser.add_argument("--repo-root",
+                      help="Directory containing sources",
+                      type=str,
+                      default=os.path.abspath(os.path.dirname(__file__)))
+  parser.add_argument("--llvm-path", help="Path to llvm-project sources")
+  parser.add_argument("--iree-path", help="Path to IREE (used if enabled)")
+  parser.add_argument("--no-ccache",
+                      help="Disables ccache (if available)",
+                      dest="enable_ccache",
+                      action="store_false",
+                      default=True)
+  parser.add_argument("--build-dir",
+                      help="Build directory",
+                      type=str,
+                      default="build")
+  return parser.parse_args()
+
+
+def main(args):
+  print(f"-- Python version {sys.version} ({sys.executable})")
+  print(f"CCACHE = {args.enable_ccache}")
+
+  llvm_path = None
+  llvm_projects = ["sandbox"]
+  llvm_configure_args = [
+      f"-DLLVM_EXTERNAL_SANDBOX_SOURCE_DIR={args.repo_root}",
+  ]
+
+  # Detect IREE (defaults LLVM path as well).
+  iree_path = args.iree_path
+  if iree_path:
+    iree_path = os.path.abspath(iree_path)
+    print(f"-- Enabling IREE from {iree_path}")
+    if not os.path.exists(os.path.join(iree_path, "CMakeLists.txt")):
+      print(f"ERROR: Could not find iree at {iree_path}")
+      return 1
+    llvm_path = os.path.join(iree_path, "third_party", "llvm-project")
+    iree_dialects_path = os.path.join(iree_path, "llvm-external-projects",
+                                      "iree-dialects")
+    if not os.path.exists(os.path.join(iree_dialects_path, "CMakeLists.txt")):
+      print(f"ERROR: Cannot find iree-dialects project at {iree_dialects_path}")
+      return 1
+    # Must come before the sandbox project.
+    llvm_projects.insert(0, "iree_dialects")
+    #llvm_projects.append("iree_dialects")
+    llvm_configure_args.append(
+        f"-DLLVM_EXTERNAL_IREE_DIALECTS_SOURCE_DIR={iree_dialects_path}")
+
+  # Detect LLVM.
+  if args.llvm_path:
+    llvm_path = os.path.abspath(args.llvm_path)
+    print(f"-- Using explicit llvm-project path: {llvm_path}")
+  elif llvm_path:
+    print(f"-- Using inferred llvm-project path: {llvm_path}")
+  else:
+    llvm_path = os.path.join(args.repo_root, "..", "llvm-project")
+    print("-- Using default llvm-project path: {llvm_path}")
+  if not os.path.exists(llvm_path):
+    print(f"ERROR: Could not find llvm-project at {llvm_path}")
+    return 1
+
+  # Detect clang.
+  clang_path = shutil.which("clang")
+  clangpp_path = shutil.which("clang++")
+  has_clang = False
+  if clang_path and clangpp_path:
+    llvm_configure_args.append(f"-DCMAKE_C_COMPILER={clang_path}")
+    llvm_configure_args.append(f"-DCMAKE_CXX_COMPILER={clangpp_path}")
+    has_clang = True
+  else:
+    print(
+        "WARNING: Could not find clang. Building with default system compiler")
+
+  # Detect lld.
+  lld_path = shutil.which("ld.lld")
+  if lld_path:
```

Comment:
should we warn if building without LLD or CCACHE?
I wasn't personally versed with those and now I'd almost consider it a bug not using it.

I usually just go locally with the following in the cmake config:
`-DLLVM_ENABLE_LLD=ON -DLLVM_CCACHE_BUILD=ON`

do we envision people need to bring their own ?

Reply:
There are... strong opinions and variation in the field. For a project like this, probably better to err on the side of user friendly defaults.